src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMapping.java

```diff
@@ -140,18 +150,51 @@ public static String getIndividualLastName(Entity entity) {
+                                                                        HearingSubtype hearingSubtype,
+                                                                        HearingOptions hearingOptions) {
+        if (hearingType == null || hearingSubtype == null) {
+            return Optional.empty();
+        }
+
+        return shouldPreferNotAttendingHearingChannel(hearingType, hearingOptions) ? Optional.ofNullable(NOT_ATTENDING.getHmcReference())
+            : isYes(hearingSubtype.getWantsHearingTypeFaceToFace()) ? Optional.ofNullable(FACE_TO_FACE.getHmcReference())
+            : shouldPreferVideoHearingChannel(hearingSubtype) ? Optional.ofNullable(VIDEO.getHmcReference())
+            : shouldPreferTelephoneHearingChannel(hearingSubtype) ? Optional.ofNullable(TELEPHONE.getHmcReference())
+            : Optional.empty();
     }
 
-    public static String getIndividualInterpreterLanguage(HearingOptions hearingOptions) {
-        // TODO Depends on SSCS-10273 - Needs to implement for Reference data to convert from SSCS Languages/Sign Languages to Reference languages
-        // if (isYes(hearingOptions.getLanguageInterpreter())) {
-        //     String signLanguageType = hearingOptions.getSignLanguageType();
-        //     String languages = hearingOptions.getLanguages();
-        // }
-        return null;
+    private static boolean shouldPreferNotAttendingHearingChannel(String hearingType, HearingOptions hearingOptions) {
+        return HEARING_TYPE_PAPER.equals(hearingType) || !hearingOptions.isWantsToAttendHearing();
+    }
+
+    private static boolean shouldPreferTelephoneHearingChannel(HearingSubtype hearingSubtype) {
+        return isYes(hearingSubtype.getWantsHearingTypeTelephone()) && nonNull(hearingSubtype.getHearingTelephoneNumber());
+    }
+
+    private static boolean shouldPreferVideoHearingChannel(HearingSubtype hearingSubtype) {
+        return isYes(hearingSubtype.getWantsHearingTypeVideo())
+            && nonNull(hearingSubtype.getHearingVideoEmail());
+    }
+
+    public static Optional<String> getIndividualInterpreterLanguage(HearingOptions hearingOptions) {
+        if (isTrue(hearingOptions.wantsSignLanguageInterpreter())) {
+            return getSignLanguage(hearingOptions)
+                .map(SignLanguage::getHmcReference);
+        }
+        if (isYes(hearingOptions.getLanguageInterpreter())) {
+            return getInterpreterLanguage(hearingOptions)
+                .map(InterpreterLanguage::getHmcReference);
+        }
+        return Optional.empty();
+    }
+
+    private static Optional<SignLanguage> getSignLanguage(HearingOptions hearingOptions) {
+        return Optional.ofNullable(SignLanguage.getSignLanguageKeyByCcdReference(hearingOptions.getSignLanguageType()));
+    }
+
+    private static Optional<InterpreterLanguage> getInterpreterLanguage(HearingOptions hearingOptions) {
+        return Optional.ofNullable(InterpreterLanguage.getLanguageAndConvert(hearingOptions.getLanguages()));
```

Comment:
Same here, can you return a String and add orElse(null) at the end.

Reply:
Done